app/src/main/java/com/arielwang/workoutlogger/features/track/ui/screen/TrackViewModel.kt

```diff
@@ -38,10 +52,88 @@ class TrackViewModel @Inject constructor(
             is TrackView.Action.GoBackToPreviousPage -> {
                 navigator.navigateUp()
             }
+            is TrackView.Action.OnMinusCounterClick -> {
+                when(action.textFieldInWhichSection) {
+                    TrackTextFieldInWhichSection.WEIGHTTEXTFIELD -> {
+                        viewState = viewState.copy(
+                            weightNumber = (viewState.weightNumber.toDouble() - 1).toString()
```

Comment:
Why here uses toDouble but the repsNumber uses toInt?

Reply:
Because the weight number can be double: 56.8 kg and the reps number represents how many times the user does the exercise.